lib/spack/spack/test/cmd/external.py

```diff
@@ -242,3 +242,15 @@ def test_new_entries_are_reported_correctly(
     # has been found
     output = external('find', 'gcc')
     assert 'No new external packages detected' in output
+
+
+def test_use_tags_for_detection(mock_executable, mutable_config, monkeypatch):
+    # Prepare an environment to detect a fake cmake
```

Comment:
You should also add a fake non-dev-tools package to the environment and make sure it does not get detected

Reply:
Done